### PR TITLE
Job manager logger file descriptor leak fixed

### LINKED ISSUE #47389: Core: File descriptor of ray job does not get closed
### What happened + What you expected to happen

File descriptor does not get closed in the job manager process on this line when long running job cluster keep receiving job. https://github.com/ray-project/ray/blob/e164d87a49f8b5c7bc5778859873d75845705b47/python/ray/dashboard/modules/job/job_manager.py#L91 leading to high cpu usage 

[
Screenshot 2024-08-28 at 8 08 38 PM
](url)

### Versions / Dependencies

Ray 2.34
Python 3.11

### Reproduction script

Ray jobs on a cluster with some fixed interval (e.g: 30 seconds)

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by ujjawal-khare-27**
## Related issue number
Closes https://github.com/ray-project/ray/issues/47389

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/dashboard/modules/job/job_manager.py [hunk 1]
```diff
@@ -505,6 +506,8 @@ async def submit_job(
                 "Please use a different submission_id."
             )
 
+        driver_logger = self._get_job_driver_logger(submission_id)
+        driver_logger.info("Runtime env is setting up.")
```
Comment (GeneDer) [inline python/ray/dashboard/modules/job/job_manager.py:510]: We can setup the logger early, but i think the `driver_logger.info("Runtime env is setting up.")` line should moved back right before supervisor was constructed as that's where the runtime env is starting to get setting up.

File: python/ray/dashboard/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import logging
+import sys
+
+import pytest
+
+from ray.dashboard.utils import close_logger_file_descriptor
+
+
+def test_close_logger_file_descriptor():
```
Comment (GeneDer) [inline python/ray/dashboard/tests/test_utils.py:9]: Thanks for writing a test👍

**Review by GeneDer — COMMENTED** [review]
Kind nit, would prefer to move back where the logging statement was. But besides that the rest of the PR LGTM!

File: python/ray/dashboard/modules/job/job_manager.py (hunk 1 above)
Comment (ujjawal-khare-27) [reply python/ray/dashboard/modules/job/job_manager.py]: Make sense. Reverted the change.

**Review by GeneDer — APPROVED** [review]
Let's go! @jjyao This PR LGTM now

**Comment by ujjawal-khare-27** [comment]
Linking [PR](https://github.com/ray-project/ray/pull/48003) for discussion

**Comment by ujjawal-khare** [comment]
@GeneDer all checks are passing. Can we please merge this PR?

**Comment by GeneDer** [comment]
I don't have permission to do that unfortunately

**Comment by ujjawal-khare-27** [comment]
@anyscalesam please let me know if anything is needed from my side.

**Comment by ujjawal-khare-27** [comment]
@rkooo567 can you please review this PR?

**Review by rkooo567 — APPROVED** [review]
LGTM!

**Pull request merged**